src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/RuntimeNodeWriter.cs

```diff
@@ -34,9 +33,14 @@ public override void WriteUsingDirective(CodeRenderingContext context, UsingDire
     {
         if (node.Source.HasValue)
         {
-            using (context.CodeWriter.BuildLinePragma(node.Source.Value, context))
+            using (context.CodeWriter.BuildEnhancedLinePragma(node.Source.Value, context, characterOffset: 1))
```

Comment:
Consider creating a tracking issue for the characterOffset = 0 bug so we don't forget about it.

Reply:
Hmm, I realized while working on a different PR that this is also wrong. In an enhanced line directive, the offset is _optional_ https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/enhanced-line-directives#detailed-design 

So, to have an effective value of `0` you have to omit the value altogether, physically writing `0` is invalid, and unlike the other values it's *not* `1`-based. Will fix.